account-kit/signer/src/base.ts

```diff
@@ -93,10 +87,12 @@ export type SignerConfig = {
  * Implements the `SmartAccountAuthenticator` interface and handles various signer events.
  */
 export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
+  extends SmartAccountSignerFromClient
```

Comment:
Wait, this is an abstract class extending a concrete class? I think this is really asking for trouble. I'm going to go into my screed against subclassing, just a moment.

Reply:
tbh feeling like adding this `SmartAccountSignerFromClient` might just be an over-abstraction.

i was trying to avoid some code duplication for handling the signMessage/signTypedData/signAuthorization, but doing all of this honestly results in more code, b/c those methods are only a couple of lines anyways.

pushing some changes to simplify.